### PR TITLE
🤖 Update GitHub release workflow

### PR DESCRIPTION
**Pull request by arybolovlev**
### Description

This PR updates the Release GitHub action workflow to pick up the latest improvements.

### Acceptance tests
- [ ] Have you added an acceptance test for the functionality being added?
- [ ] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'
...
```

### Release Note

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/CHANGELOG.md):

```release-note
NONE
```

### References

N/A

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -9,14 +9,6 @@ permissions:
...
 
 jobs:
-  go-version:
-    runs-on: ubuntu-latest
-    outputs:
-      version: ${{ steps.go-version.outputs.version }}
-    steps:
-      - uses: actions/checkout@v2
-      - id: go-version
-        run: echo "::set-output name=version::$(cat ./.go-version)"
```
Comment (BBBmau) [inline .github/workflows/release.yaml:19]: Are we required to remove the `actions/checkout` job? I thought we needed this is in our workflow to access the repo under `$GITHUB_WORKSPACES`

Comment (BBBmau) [reply .github/workflows/release.yaml:19]: I saw that this is replaced with the pinned commit hash for checkout in #1961. Looks good!

**Review by BBBmau — APPROVED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (arybolovlev) [reply .github/workflows/release.yaml:19]: In #1961 I made bulk blind updates for all workflow files. :)

This one is more specific to the latest changes in the action. `go-version` is one of the jobs that only reads the Golang version and pass the result to jobs down the road. With the latest version of the action `hashicorp/ghaction-terraform-provider-release` we can rid of this step. Once it is done we should be able to rid of `.go-version` file and rely on `go.mod` in all out actions.

**Review by alexsomesan — APPROVED** [review]
LGTM!

**Pull request merged**